src/write.rs

```diff
@@ -1858,7 +1777,7 @@ fn write_central_directory_header<T: Write>(writer: &mut T, file: &ZipFileData)
         writer.write_all(central_extra_field)?;
     }
     // file comment
-    // <none>
+    writer.write_all(file.file_comment.as_bytes())?;
```

Comment:
@Pr0methean Note that we are changing the behavior here to actually write the file comment, which I believe is correct, and at a quick glance, I assumed that the previous code would have produced an invalid zip file for non-empty comments. Whether or not the previous code was broken, I want to make it clear that this is a big change in output behavior. I would prefer to leave the code here as-is and then make a followup PR adding tests for file comments -- does that sound reasonable?

Reply:
Since this PR won't be ready for release without those tests, and will need a major-version bump either way, I've created a new branch and set that branch as this PR's target. It can be merged into mainline once the tests are also merged. In either case I expect `release-plz` will decide a major version bump is needed.